spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/mail/MailSenderAutoConfiguration.java

```diff
@@ -40,6 +41,7 @@
  */
 @Configuration
 @ConditionalOnClass({MailSender.class, AmazonSimpleEmailService.class})
+@ConditionalOnMissingBean(MailSender.class)
```

Comment:
Hello, @candrews !

Thank you for the PR, but to be honest I don't like this approach.
Although I see that it is typical in Spring Boot. For example similar `org.springframework.boot.autoconfigure.mail.MailSenderAutoConfiguration` has the same `@ConditionalOnMissingBean(MailSender.class)`.

So, right now, when I have both these configuration in class path, the regular `JavaMailSenderImpl` `@Bean` is overridden by this SES `MailSender`.

Please, describe how would you deal if we need both `MailSender`s? For Google and AWS, for example? Or that does not make sense and we really should use only one?

From here, otherwise, we might consider something like `@AutoConfigureBefore(org.springframework.boot.autoconfigure.mail.MailSenderAutoConfiguration.class)` to let end-user solution win and override the Boot solution at the same time, if we win here.